### DISCUSSION TITLE
Params and Overrides

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#5243] [Q&A]
With this code:
```
var V = new Foo();

var Expected = "1-2-3-4-5";
var Actual = V.ToString();

if(Expected != Actual) {
    Console.WriteLine("ERROR");
    Console.WriteLine($@"Expected: {Expected}");
    Console.WriteLine($@"Actual:   {Actual}");
}


public record Foo {

    public string ToString(params int[] Values) {
        var ret = string.Join("-", Values.Select(x => x.ToString()));

        return ret;
    }

    public override string ToString() {
        return ToString(1, 2, 3, 4, 5);
    }
}
```

Why does ```.ToString()``` resolve to ```public string ToString(params int[] Values)``` and not ```public override string ToString()```?

**Comment by 333fred** [comment]
Because of this section of the specification: https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/expressions.md#125-member-lookup. Specifically, new members (overridden members are not considered new, but your params overload is a new member) are considered first, before members from a base type.
